### DISCUSSION TITLE
Using Refit/Named Clients with SentryHttpMessageHandler

### DISCUSSION THREAD

**Discussion by SteveSitekitcare** [getsentry/sentry-dotnet#2735] [Q&A]
Hey all,

I'm tyring to setup Sentry for MAUI. Im using dotnet 7.
My application sets up all its httpclients using named clients with refit.
I cant see how refit is an issue here as it simply registers the same way as the default.
I'm producing 504 status codes which are withing the 500 range that docs state is the default to track
e.g.
DEFAULT
   builder.Services.AddHttpClient<GitHubService>();
REFIT
   builder.Services.AddRefitClient<GitHubService>()

So my sample code looks like this 
```
builder.Services.AddTransient<SentryHttpMessageHandler>();

builder.Services.AddRefitClient<IProfileClient>()
    .ConfigureHttpClient(c => c.BaseAddress = new Uri(MyAPIURL))
    .AddHttpMessageHandler<SentryHttpMessageHandler>()
```

This isnt producing anything in sentry

the docs do state the following
https://docs.sentry.io/platforms/dotnet/guides/maui/configuration/http-client-errors/?original_referrer=https%3A%2F%2Fdocs.sentry.io%2F
specifically this section
![image](https://github.com/getsentry/sentry-dotnet/assets/138577988/9443dd76-c681-4574-8c5e-f9287e81a95b)

Im registering using the Sentry Maui extension as the setup doc describes 
https://docs.sentry.io/platforms/dotnet/guides/maui/configuration/http-client-errors/?original_referrer=https%3A%2F%2Fdocs.sentry.io%2F

I've even asked the bot. I cant seem to get any errors

**Comment by bitsandfoxes** [comment]
Hey there! Let me give that a try and get back to you.

**Reply by SteveSitekitcare** [reply]
Hi @bitsandfoxes

Is there any update?

**Reply by bitsandfoxes** [reply]
Hey there. Sorry for the delay, I'm struggling a bit to even get a MAUI sample running locally. But what I found so far was that C# `SentryHttpMessageHandler` runs on top of the Cocoa SDK integration that the SDK comes with to deal with everything native. We have a similar issue [here](https://github.com/getsentry/sentry-dotnet/issues/2744) that is like the inverse of your problem: The C# filtering does not take.
I'm still digging into this further.

**Reply by bitsandfoxes** [reply]
So I managed to recreate this and "it works for me". Can you tell me about how you retrieved the refit client?
I noticed that `RestService.For<>();`did not work for me but the injected one does work.
Too good actually, since we're generating multiple events right now. One from the MessageHandler and one from an unhandled exception.
